folder_paths.py

```diff
@@ -243,6 +243,9 @@ def map_filename(filename):
     def compute_vars(input, image_width, image_height):
         input = input.replace("%width%", str(image_width))
         input = input.replace("%height%", str(image_height))
+        input = input.replace("%year%", time.strftime("%Y"))
+        input = input.replace("%month%", time.strftime("%m"))
+        input = input.replace("%day%", time.strftime("%d"))
         return input
 
     filename_prefix = compute_vars(filename_prefix, image_width, image_height)
```

Comment:
i'd suggest to add a guard here for if prefix contains `%`, so it doesn't do redundant processing if there's no vars anyway

Reply:
I didn't really get this part, would I be guarding against all? something like this

`    if "%year%" not in input and "%month%" not in input and "%day%" not in input:
        return input`